src/realm/query_engine.hpp

```diff
@@ -1661,13 +1673,37 @@ class OrNode : public ParentNode {
         return s;
     }
 
-
     void init() override
     {
         ParentNode::init();
 
         m_dD = 10.0;
 
+        StringNode<Equal>* first = nullptr;
+        std::sort(m_conditions.begin(), m_conditions.end(),
+                  [](auto& a, auto& b) { return a->m_condition_column_idx < b->m_condition_column_idx; });
+        auto it = m_conditions.begin();
+        while (it != m_conditions.end()) {
+            // Only try to optimize on StringNode<Equal> conditions without search index
+            if (bool(*it) && (first = dynamic_cast<StringNode<Equal>*>(it->get())) && !first->has_search_index()) {
```

Comment:
Sligtly confused by this bool(*it) ... if it is necessary here, then how come we don't need it before dereferencing "next" in line 1691 ?

Perhaps some simplification is possible?

Reply:
Thanks, I removed the cast.